crowbar_framework/app/models/node_object.rb

```diff
@@ -724,7 +724,12 @@ def rebuild_run_list
     crowbar["run_list_map"] = {} if crowbar["run_list_map"].nil?
 
     # Cull by state
-    map = crowbar["run_list_map"].select { |k,v| v["states"].include?("all") or v["states"].include?(self.crowbar["state"]) }
+    node_state = self.crowbar['state']
+    map = crowbar["run_list_map"].select do |k, v|
+      v["states"].include?(node_state) ||
+      # When node is in upgrade state, allow only upgrade specific recipes
+      (v["states"].include?("all") && node_state != "crowbar_upgrade")
```

Comment:
Style/MultilineOperationIndentation: Use 2 (not 0) spaces for indenting an expression in an assignment spanning multiple lines.